Drop support for Python 3.8 and add support for Python 3.13

juergbi opened the pull request:
Python 3.8 is no longer maintained upstream and some Python packages have dropped support for it. Updating dependencies is required for Python 3.13 support but will break Python 3.8 support.

juergbi committed:
Add support for Python 3.13

juergbi committed:
Update dependencies for Python 3.13 support

juergbi committed:
Drop support for Python 3.8

Python 3.8 is no longer maintained upstream and some Python packages
have dropped support for it.

abderrahim commented:
Please check the code, we might have some python 3.8 specific codepaths that we can drop.

juergbi committed:
ci: Update Ubuntu images from 20.04 to 22.04

This drops testing of end-of-life Python 3.8 and replaces it with test
coverage of Python 3.10.

juergbi committed:
ci: Run tests also on Fedora 41

This extends test coverage to Python 3.13.

juergbi committed:
Fix type errors reported by the latest version of mypy

juergbi committed:
_testing/runcli.py: Drop workaround for AIX

mypy reports an error with this workaround:

    error: Cannot assign to final name "__stdout__"  [misc]

juergbi commented:
> Please check the code, we might have some python 3.8 specific codepaths that we can drop.

I don't see any Python 3.8-specific code paths. However, I've removed one left-over conditional from Python 3.7.

juergbi committed:
ci: Update images

https://gitlab.com/BuildStream/buildstream-docker-images/-/merge_requests/225

juergbi committed:
tests/remotecache/simple.py: Support new gRPC in error message check
